spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/ExpireSnapshotsProcedure.java

```diff
@@ -117,6 +122,12 @@ public InternalRow[] call(InternalRow args) {
             action.executeDeleteWith(executorService(maxConcurrentDeletes, "expire-snapshots"));
           }
 
+          if (wapId != null) {
+            Iterables.filter(
+                    table.snapshots(), snapshot -> wapId.equals(WapUtil.stagedWapId(snapshot)))
+                .forEach(snapshot -> action.expireSnapshotId(snapshot.snapshotId()));
```

Comment:
I'm not sure about this. I expect the procedures to expose similar options as the table operations, but this is actually creating an option that doesn't exist in `expireSnapshots`. What about a more consistent way to do this? We could allow passing an array of snapshot IDs, for example, that would bake down to `expireSnapshotId` calls like this.

What do you think?

Reply:
@rdblue  That sounds reasonable, that way we can keep this procedure agnostic of `wap_id`. 

We can then use this procedure after finding the snapshot ID of the wap write we are performing. Let me do the changes.

Reply:
@rdblue done, I've modified this PR to pass an array of snapshot IDs, please take a look.